### DISCUSSION TITLE
Howto solve CORS?

### DISCUSSION THREAD

**Discussion by pherk** [Jinntec/Fore#68] [Q&A] [answered]
Browser complain about CORS configuration if submission url refers to a foreign server.
es-dev-server has an option for CORS which can be set to 'true' but it doesn't solve the issue.

I found a solution by proxying:

- https://www.thisdot.co/blog/how-to-solve-cors-using-a-proxy-server-and-koa

1. install koa libaries:  npm i koa & npm i @koa/cors koa-proxies
2. change es-dev-server.json
~~~
    +  const proxy = require('koa-proxies');
    ...
    +     cors : true,      
    +     middlewares: [
    +       proxy('/base-dir', {
    +           target: 'htp://third-party-ip/',    //base-dir will be added to path
    +           changeOrigin: true
    +       }),
    +     ],
~~~
3. submission via base-dir
 ~~~
   <fx-submission id="search"
       url="/base-dir/path?query-string"
       method="get"
       replace="instance"
       instance="search-result">
   </fx-submission>
~~~

**Comment by JoernT** [comment, marked as answer]
This was already answered by the author :)

**Reply by pherk** [reply]
One (small) problem remains, as I discovered today.
If the proxy call returns a HTTP code other than 200, then a pseudo CORS error will occur, which prevents own error handling.
